.github/workflows/foreman_plugin.yml

```diff
@@ -97,12 +104,39 @@ jobs:
         uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node }}
+      - name: Setup NPM Cache
+        if: ${{ hashFiles(format('{0}/package.json', inputs.plugin)) != '' && !contains(matrix.task, 'seed') }}
+        uses: actions/cache@v3
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ matrix.node }}-${{ hashFiles('**/package.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-${{ matrix.node }}-
+      - name: "Build package-lock.json"
+        if: ${{ hashFiles(format('{0}/package.json', inputs.plugin)) != '' && !contains(matrix.task, 'seed') }}
+        run: bundle exec npm install --package-lock-only --no-audit
+      - name: Archive package-lock.json
+        if: ${{ hashFiles(format('{0}/package.json', inputs.plugin)) != '' && !contains(matrix.task, 'seed') }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: package-lock-ruby-${{ matrix.ruby }}-node-${{ matrix.node }}-pg-${{ matrix.postgresql }}.json
+          path: package-lock.json
+      - name: "Install NPM packages"
+        if: ${{ hashFiles(format('{0}/package.json', inputs.plugin)) != '' && !contains(matrix.task, 'seed') }}
+        run: bundle exec npm ci --no-audit
       - name: Prepare test env
         run: |
           bundle exec rake db:create
           bundle exec rake db:migrate
-      - name: Run plugin tests
-        run: bundle exec rake test:${{ inputs.plugin }}
+      - name: "Compile webpack"
+        if: ${{ hashFiles(format('{0}/package.json', inputs.plugin)) != '' && contains(matrix.task, 'test') }}
+        run:
+          bundle exec rake webpack:compile
```

Comment:
IMHO, if `test:$plugin` depends on webpack (because it contains integration tests), it should depend on that itself via Rake, and not make us do this hack, but :woman_shrugging:

Reply:
You could argue that it can make it painful for developers. If you have webpack running then you don't need to compile, but we can't detect this. So I can accept this now. Perhaps after we merge webpack 5 we can be a bit smarter and introduce some task that only compiles if some file doesn't exist.